pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -651,3 +658,15 @@ def configure_slurm_ddp(self):
         # notify user the that slurm is managing tasks
         if self.is_slurm_managing_tasks:
             rank_zero_info("Multi-processing is handled by Slurm.")
+
+    @property
+    def _sharded_training_type(self) -> bool:
+        return isinstance(self.training_type_plugin,
+                          (DDPShardedPlugin, DDPSpawnShardedPlugin
+                           )) or self._distrib_type in (DistributedType.DDP_SHARDED, DistributedType.DDP_SHARDED_SPAWN)
+
+    @property
+    def _fully_sharded_training_type(self) -> bool:
+        return isinstance(
+            self.training_type_plugin, FullyShardedPlugin
+        ) or self._distrib_type == DistributedType.FULLY_SHARDED
```

Comment:
for later - should these be properties on the plugin? this would let more custom plugins work w/o needing to subclass the existing ones

Reply:
yep all of these should be properties, might naturally come out of #6090